File: src/main/java/com/github/joelws/release/tracker/dao/ArtifactDaoImpl.java
package com.github.joelws.release.tracker.dao;


import com.github.joelws.release.tracker.common.GenericDaoImpl;
import com.github.joelws.release.tracker.model.Artifact;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
@Repository
public class ArtifactDaoImpl extends GenericDaoImpl<Artifact, String> implements ArtifactDao
{
    public ArtifactDaoImpl()
    {
        super(Artifact.class);
    }
    @Transactional
    @SuppressWarnings(value = "unchecked") @Override public List<Artifact> findAllById(String id)
    {
        String query = String.format("SELECT a FROM Artifact a WHERE a.id = '%s'", id);
        return (List<Artifact>) super.getEntityManager().createQuery(query).getResultList();
    }
}

File: src/main/kotlin/com/joelws/release/tracker/dao/artifact/ArtifactDaoImpl.kt
package com.joelws.release.tracker.dao.artifact

import com.joelws.release.tracker.dao.GenericDaoImpl
import com.joelws.release.tracker.entity.artifact.Artifact
import com.joelws.release.tracker.entity.artifact.ArtifactPK

import org.springframework.stereotype.Repository

@Repository
open class ArtifactDaoImpl : GenericDaoImpl<Artifact, ArtifactPK>(), ArtifactDao {

    override fun list(): List<Artifact> {
        val query = "SELECT a FROM Artifact a"
        @Suppress("UNCHECKED_CAST")
        return entityManager.createQuery(query).resultList as List<Artifact>
    }
}
